src/plugins/vis_type_vega/public/vega_inspector/vega_adapter.ts

```diff
@@ -0,0 +1,137 @@
+import { map } from 'rxjs/operators';
+import { View, Runtime, Spec } from 'vega';
+import { i18n } from '@kbn/i18n';
+import { Assign } from '@kbn/utility-types';
+
+interface DebugValues {
+  view: View;
+  spec: Spec;
+}
+
+export interface VegaRuntimeData {
+  columns: Array<{
+    id: string;
+  }>;
+  data: Array<Record<string, string>>;
+}
+
+export type InspectDataSets = Assign<VegaRuntimeData, { id: string }>;
+export type InspectSignalsSets = VegaRuntimeData;
+
+/** Get Runtime Scope for Vega View
+ * @link https://vega.github.io/vega/docs/api/debugging/#scope
+ **/
+const getVegaRuntimeScope = (debugValues: DebugValues) =>
+  (debugValues.view as any)._runtime as Runtime;
+
+const serializeColumns = (item: Record<string, unknown>, columns: string[]) => {
+  const nonSerializableFieldLabel = '(..)';
+
+  return columns.reduce((row: Record<string, string>, column) => {
+    try {
+      row[column] = JSON.stringify(item[column]);
+    } catch (e) {
+      row[column] = nonSerializableFieldLabel;
+    }
+    return row;
+  }, {});
+};
+
+const mapColumns = (columns: string[]) => columns.map((column) => ({ id: column, schema: 'json' }));
+
+export class VegaAdapter {
+  private debugValuesSubject = new BehaviorSubject<DebugValues | null>(null);
+
+  bindInspectValues(debugValues: DebugValues) {
+    this.debugValuesSubject.next(debugValues);
+  }
+
+  getDataSetsSubscription(): Observable<InspectDataSets[]> {
+    return this.debugValuesSubject.pipe(
+      map((debugValues) => {
+        if (!debugValues) {
+          return [];
+        }
+
+        const runtimeScope = getVegaRuntimeScope(debugValues);
+
+        return Object.keys(runtimeScope.data || []).reduce((acc: InspectDataSets[], key) => {
+          const value = runtimeScope.data[key].values.value;
+
+          if (value && value[0]) {
+            const columns = Object.keys(value[0]);
+            acc.push({
+              id: key,
+              columns: mapColumns(columns),
+              data: value.map((item: Record<string, unknown>) => serializeColumns(item, columns)),
+            });
+          }
+          return acc;
+        }, []);
+      })
+    );
+  }
+
+  getSignalsSetsSubscription(): Observable<InspectSignalsSets | undefined> {
+    return this.debugValuesSubject.pipe(
+      map((debugValues) => {
+        if (!debugValues) {
+          return undefined;
+        }
+
+        const runtimeScope = getVegaRuntimeScope(debugValues);
+        const columns = [
+          i18n.translate('visTypeVega.inspector.vegaAdapter.signal', {
+            defaultMessage: 'Signal',
+          }),
+          i18n.translate('visTypeVega.inspector.vegaAdapter.value', {
+            defaultMessage: 'Value',
+          }),
+        ];
+
+        return {
+          columns: mapColumns(columns),
+          data: Object.keys(runtimeScope.signals).map((key: string) => {
+            return serializeColumns(
+              {
+                [columns[0]]: key,
```

Comment:
It seems like the key here is getting converting to a JSON-safe string by using double quotes, and the values are getting converted to JSON without nice formatting. Can you improve the formatting?

<img width="1056" alt="Screenshot 2020-07-15 15 03 35" src="https://user-images.githubusercontent.com/666475/87586452-8cf99f00-c6ae-11ea-97ea-341e3cec4d4d.png">

Reply:
done